samples/bot-calling-meeting/csharp/Source/CallingBotSample/Bots/CallingBot.cs

```diff
@@ -98,21 +108,69 @@ private async Task NotificationProcessor_OnNotificationReceivedAsync(Notificatio
             // https://microsoftgraph.github.io/microsoft-graph-comms-samples/docs/articles/index.html#answer-incoming-call-with-service-hosted-media
 
             graphLogger.CorrelationId = args.ScenarioId;
+            var callId = GetCallIdFromNotification(args);
+
             if (args.ResourceData is Call call)
             {
+
                 if (args.ChangeType == ChangeType.Created && call.State == CallState.Incoming)
                 {
-                    await callService.Answer(GetCallIdFromNotification(args), audioRecordingConstants.Speech, audioRecordingConstants.PleaseRecordYourMessage);
+                    await callService.Answer(callId, audioRecordingConstants.Speech, audioRecordingConstants.PleaseRecordYourMessage);
                 }
                 else if (
                     args.ChangeType == ChangeType.Updated
-                    && call.State == CallState.Established
-                    // The below helps to distinguish between two similar Established notifications that are sent for incoming 1:1 meetings
-                    && (call.MediaState?.Audio == MediaState.Active
-                    // And this covers calls that the bot creates
-                    || call.Direction == CallDirection.Outgoing))
+                    && call.State == CallState.Established)
+                {
+                    // Some scenarios fire two CallState.Established events. The use of a cache ensures we only play the prompt once on meeting join
+                    string key = $"{callId}:established";
+                    if (!callBotCache.Get<bool>(key))
+                    {
+                        callBotCache.Set(key, true);
```

Comment:
What removes entries from this cache?

Reply:
I've updated the code to set an expiration using MemoryCacheEntryOptions when adding the items to the cache